server/server.go

```diff
@@ -0,0 +1,113 @@
+package server
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"sync/atomic"
+
+	"github.com/babylonlabs-io/finality-gadget/config"
+	"github.com/babylonlabs-io/finality-gadget/db"
+	"github.com/babylonlabs-io/finality-gadget/finalitygadget"
+	"github.com/lightningnetwork/lnd/signal"
+	"go.uber.org/zap"
+
+	"google.golang.org/grpc"
+)
+
+// Server is the main daemon construct for the EOTS manager server. It handles
+// spinning up the RPC sever, the database, and any other components that the
+// EOTS manager server needs to function.
+type Server struct {
+	rpcServer *rpcServer
+	cfg       *config.Config
+	db        db.IDatabaseHandler
+
+	logger *zap.Logger
+
+	quit        chan struct{}
```

Comment:
Is this used and do we provide graceful shutdown support in this PR?

From what I see `startGrpcListen` seems to not subscribe to any quit channel

Reply:
indeed this is not used, i will remove it

btw it is copied from https://github.com/babylonlabs-io/finality-provider/blob/main/eotsmanager/service/server.go so we should also amend there